tvm-tac/src/main/kotlin/org/ton/tac/TacGeneration.kt

```diff
@@ -302,49 +346,274 @@ private fun <Inst : AbstractTacInst> processOrdinaryInst(
             endingInstGenerator to null
         }
 
+    return ControlFlowPreparation(
+        outputVars = outputVars,
+        newEndInstGenerator = newEndInstGenerator,
+        label = label,
+        inputVars = inputVars,
+    )
+}
+
+fun tupleCompatible(
```

Comment:
All logic related to compatibility checks is better to move to a separate file (for example, `TacCompatibilityChecks.kt`).